src/NServiceBus.Core/EndpointCreator.cs

```diff
@@ -91,9 +89,13 @@ void Initialize()
 
             recoverabilityComponent = new RecoverabilityComponent(settings);
 
+            featureComponent = new FeatureComponent(settings);
+
             var featureConfigurationContext = new FeatureConfigurationContext(settings, containerComponent.ContainerConfiguration, pipelineComponent.PipelineSettings, routingComponent, receiveConfiguration);
-            var featureStats = featureActivator.SetupFeatures(featureConfigurationContext);
-            settings.AddStartupDiagnosticsSection("Features", featureStats);
+
+            //note: This is where the settings gets locked since the feature component (unfortunately) uses the settings to store feature state.
+            // Locking happens just before the Features gets "setup"
+            featureComponent.Initalize(concreteTypes, containerComponent, featureConfigurationContext);
```

Comment:
afaik we could move the locking of the settings after the call to `SetupFeatures`. This is a change I've committed as part of a refactoring before but ended up closing the PR for other reasons. Looking at the current implementation, I have a hunch that we could lock the settings after the component.Initialize which would be a +100 from my side. This logic should absolutely not be inside that component.

Reply:
👍 Eagerly awaits your PR 🙇